Gems/Atom/Utils/Code/Include/Atom/Utils/StableDynamicArray.inl

```diff
@@ -66,46 +66,35 @@ namespace AZ
     auto StableDynamicArray<T, ElementsPerPage, Allocator>::emplace(Args&& ... args)->Handle
     {
         // Try to find a page we can fit this in.
-        while (m_firstAvailablePage)
+        for (m_firstAvailablePage; m_firstAvailablePage < m_pages.size(); ++m_firstAvailablePage)
         {
-            size_t pageItem = m_firstAvailablePage->Reserve();
-            if (pageItem != Page::InvalidPage)
+            if (m_pages[m_firstAvailablePage])
             {
-                // pageItem is a valid item that's been reserved, so construct a new T on it.
-                T* item = m_firstAvailablePage->GetItem(pageItem);
-                AZStd::Internal::construct<T*>::single(item, AZStd::forward<Args>(args) ...);
+                size_t pageItem = m_pages[m_firstAvailablePage]->Reserve();
+                if (pageItem != Page::InvalidPage)
+                {
+                    // pageItem is a valid item that's been reserved, so construct a new T on it.
+                    T* item = m_pages[m_firstAvailablePage]->GetItem(pageItem);
+                    AZStd::Internal::construct<T*>::single(item, AZStd::forward<Args>(args)...);
 
-                ++m_itemCount;
-                return Handle(item, m_firstAvailablePage);
-            }
-            if (!m_firstAvailablePage->m_nextPage)
-            {
-                // no more pages, break and make a new one.
-                break;
+                    ++m_itemCount;
+                    return Handle(
+                        item, m_pages[m_firstAvailablePage], static_cast<uint16_t>(m_firstAvailablePage), static_cast<uint16_t>(pageItem));
+                }
             }
-            m_firstAvailablePage = m_firstAvailablePage->m_nextPage;
         }
 
         // No page to emplace in, so make a new page
         Page* page = AddPage();
-        if (m_firstAvailablePage)
-        {
-            m_firstAvailablePage->m_nextPage = page;
-        }
-        else
-        {
-            // If m_firstAvailablePage was nullptr, then there were no pages so m_firstPage would also be null, and needs to be set to the new page.
-            m_firstPage = page;
-        }
 
         // A new page was created since there was no room in any other page, so this new page will also be the first page where slots are available.
-        m_firstAvailablePage = page;
+        m_firstAvailablePage = m_pages.size() - 1;
 
         size_t pageItem = page->Reserve();
-        T* item = m_firstAvailablePage->GetItem(pageItem);
+        T* item = m_pages[m_firstAvailablePage]->GetItem(pageItem);
         AZStd::Internal::construct<T*>::single(item, AZStd::forward<Args>(args) ...);
         ++m_itemCount;
-        return Handle(item, page);
+        return Handle(item, page, static_cast<uint16_t>(m_firstAvailablePage), static_cast<uint16_t>(pageItem));
```

Comment:
nit: seems like we could use a small inline constuctEntry function to hold the duplicated code.